src/Listener/JsonApiListener.php

```diff
@@ -483,6 +483,12 @@ protected function _getEntityList($entityName, $associations)
      */
     protected function _checkRequestData()
     {
+        // prevent false positives for GET requests using query parameters
+        // as those will also have request data
```

Comment:
Just tested again and I am now certain that it is pagination which is filling the request data. Changed the comment in the code accordingly. Thanks for checking and.. apologies for driving you crazy 😆 If you disagree let me know.

```php
    public function beforeFilter(Event $event)
    {
        parent::beforeFilter($event);
        $this->paginate = [
            'sortWhitelist' => [
                'code',
                'name'
            ]
        ];
    }
``